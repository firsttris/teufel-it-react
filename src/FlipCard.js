import React, { Component } from 'react';

const cardStyle = {
  position: 'absolute',
  top: '0',
  width: '100%',
  backfaceVisibility: 'hidden'
}
const borderStyle = {
  border: '1px solid #f00',
  borderRadius: '5px',
  width: '50%',
  margin: 'auto',
  height: '175px',
  display: 'flex',
  alignItems: 'center'
}
const objectPerspectiveStyle = { perspective: '800px', 'height': '250px' }
const iconFontStyle = { fontSize: '10em', color: '#f00' }

class FilpCard extends Component {
  constructor() {
    super();
    this.state = {
      flipped: false
    }
  }

  onMouseLeave() {
    this.setState({ flipped: false })
  }

  onMouseEnter() {
    this.setState({ flipped: true })
  }

  onClick() {

  }

  render() {
    const flipStyle = {
      width: '100%',
      transformStyle: 'preserve-3d',
      transition: 'transform 1s linear',
      backfaceVisibility: 'hidden',
      transform: 'rotateY(0deg)'
    }
    flipStyle.transform = this.state.flipped ? 'rotateY(180deg)' : 'rotateY(0deg)';
    return (
      <div className="col-xs-12 col-md-6 col-lg-3" style={objectPerspectiveStyle}
        onMouseLeave={this.onMouseLeave.bind(this)}
        onMouseEnter={this.onMouseEnter.bind(this)}
        onClick={this.onClick.bind(this)}>
        <div style={flipStyle}>
          <div style={cardStyle}>
            <i className={"fa fa-5x " + this.props.icon} aria-hidden="true" style={iconFontStyle}></i>
            <div className="text-uppercase" style={{ fontSize: '1.5em' }}>{this.props.title}</div>
          </div>
          <div style={{ ...cardStyle, ...{ transform: 'rotateY(180deg)' } }}>
            <div className="p-2" style={borderStyle}>{this.props.text}</div>
          </div>
        </div >
      </div >
    );
  }
}

export default FilpCard;
