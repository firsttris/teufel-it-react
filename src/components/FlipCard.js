import React, {Component} from 'react'
import {FormattedMessage} from 'react-intl'

const cardStyle = {
    position: 'absolute',
    top: '0',
    width: '100%',
    backfaceVisibility: 'hidden'
};
const borderStyle = {
    border: '1px solid #f00',
    borderRadius: '5px',
    width: '50%',
    margin: 'auto',
    height: '175px',
    display: 'flex',
    alignItems: 'center'
};
const objectPerspectiveStyle = {perspective: '800px', 'height': '250px'}
const iconFontStyle = {fontSize: '10em', color: '#f00'}

export default class FilpCard extends Component {
    constructor() {
        super();
        this.state = {
            flipped: false
        }
    }

    onMouseLeave() {
        this.setState({flipped: false})
    }

    onMouseEnter() {
        this.setState({flipped: true})
    }

    onClick() {

    }

    render() {
        const flipStyle = {
            width: '100%',
            transformStyle: 'preserve-3d',
            transition: 'transform 1s linear',
            backfaceVisibility: 'hidden',
            transform: 'rotateY(0deg)'
        }
        flipStyle.transform = this.state.flipped ? 'rotateY(180deg)' : 'rotateY(0deg)';
        return (
            <div className={this.props.class} style={objectPerspectiveStyle}
                 onMouseLeave={this.onMouseLeave.bind(this)}
                 onMouseEnter={this.onMouseEnter.bind(this)}
                 onClick={this.onClick.bind(this)}>
                <div style={flipStyle}>
                    <div style={cardStyle}>
                        <i className={"fa fa-5x " + this.props.icon} aria-hidden="true" style={iconFontStyle}/>
                        <div className="text-uppercase" style={{fontSize: '1.5em'}}>
                            <FormattedMessage id={this.props.title}/>
                        </div>
                    </div>
                    <div style={{...cardStyle, ...{transform: 'rotateY(180deg)'}}}>
                        <div className="p-2" style={borderStyle}>
                            <FormattedMessage id={this.props.text}/>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
