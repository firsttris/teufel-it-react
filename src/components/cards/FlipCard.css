.cards {
    backface-visibility: hidden;
    -webkit-backface-visibility: hidden;
}