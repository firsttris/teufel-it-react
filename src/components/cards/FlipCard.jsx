import React, { Component } from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import './FlipCard.css';

const cardStyle = {
  position: 'absolute',
  top: '0',
  width: '100%'
};
const borderStyle = {
  border: '1px solid #f00',
  borderRadius: '5px',
  width: '50%',
  margin: 'auto',
  height: '175px',
  display: 'flex',
  alignItems: 'center'
};
const objectPerspectiveStyle = { perspective: '800px', height: '250px' };
const iconFontStyle = { fontSize: '10em', color: '#f00' };

class FlipCard extends Component {
  constructor() {
    super();
    this.state = {
      flipped: false
    };
  }

  onMouseLeave() {
    this.setState({ flipped: false });
  }

  onMouseEnter() {
    this.setState({ flipped: true });
  }

  onClick() {
    this.setState({ flipped: !this.state.flipped });
  }

  render() {
    const flipStyle = {
      width: '100%',
      transformStyle: 'preserve-3d',
      transition: 'transform 1s linear',
      transform: 'rotateY(0deg)'
    };
    flipStyle.transform = this.state.flipped ? 'rotateY(180deg)' : 'rotateY(0deg)';
    return (
      <div
        className={this.props.class}
        style={objectPerspectiveStyle}
        onMouseLeave={() => this.onMouseLeave()}
        onMouseEnter={() => this.onMouseEnter()}
        onClick={() => this.onClick()}
      >
        <div style={flipStyle}>
          <div className="cards" style={cardStyle}>
            <i className={'fa fa-5x ' + this.props.icon} aria-hidden="true" style={iconFontStyle} />
            <div className="text-uppercase" style={{ fontSize: '1.5em' }}>
              <FormattedMessage id={this.props.title} />
            </div>
          </div>
          <div className="cards" style={{ ...cardStyle, ...{ transform: 'rotateY(180deg)' } }}>
            <div className="p-2" style={borderStyle}>
              <FormattedMessage id={this.props.text} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

FlipCard.propTypes = {
  class: PropTypes.string,
  icon: PropTypes.string,
  title: PropTypes.string,
  text: PropTypes.string
};

export default FlipCard;
