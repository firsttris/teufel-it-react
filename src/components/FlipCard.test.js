import React from 'react';
import FlipCard from './FlipCard';
import renderer from 'react-test-renderer';

test('FlipCard changes the class when hovered', () => {
  const component = renderer.create(
    <FlipCard key="1" class="col-xs-12 col-md-6 col-lg-3" icon="fa-github" title="Hello" text="test"></FlipCard>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});