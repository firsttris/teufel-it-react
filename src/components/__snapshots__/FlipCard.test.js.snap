// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlipCard changes the class when hovered 1`] = `
<div
  className="col-xs-12 col-md-6 col-lg-3"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "height": "250px",
      "perspective": "800px",
    }
  }
>
  <div
    style={
      Object {
        "backfaceVisibility": "hidden",
        "transform": "rotateY(0deg)",
        "transformStyle": "preserve-3d",
        "transition": "transform 1s linear",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "backfaceVisibility": "hidden",
          "position": "absolute",
          "top": "0",
          "width": "100%",
        }
      }
    >
      <i
        aria-hidden="true"
        className="fa fa-5x fa-github"
        style={
          Object {
            "color": "#f00",
            "fontSize": "10em",
          }
        }
      />
      <div
        className="text-uppercase"
        style={
          Object {
            "fontSize": "1.5em",
          }
        }
      >
        Hello
      </div>
    </div>
    <div
      style={
        Object {
          "backfaceVisibility": "hidden",
          "position": "absolute",
          "top": "0",
          "transform": "rotateY(180deg)",
          "width": "100%",
        }
      }
    >
      <div
        className="p-2"
        style={
          Object {
            "alignItems": "center",
            "border": "1px solid #f00",
            "borderRadius": "5px",
            "display": "flex",
            "height": "175px",
            "margin": "auto",
            "width": "50%",
          }
        }
      >
        test
      </div>
    </div>
  </div>
</div>
`;

exports[`FlipCard changes the class when hovered 2`] = `
<div
  className="col-xs-12 col-md-6 col-lg-3"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "height": "250px",
      "perspective": "800px",
    }
  }
>
  <div
    style={
      Object {
        "backfaceVisibility": "hidden",
        "transform": "rotateY(180deg)",
        "transformStyle": "preserve-3d",
        "transition": "transform 1s linear",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "backfaceVisibility": "hidden",
          "position": "absolute",
          "top": "0",
          "width": "100%",
        }
      }
    >
      <i
        aria-hidden="true"
        className="fa fa-5x fa-github"
        style={
          Object {
            "color": "#f00",
            "fontSize": "10em",
          }
        }
      />
      <div
        className="text-uppercase"
        style={
          Object {
            "fontSize": "1.5em",
          }
        }
      >
        Hello
      </div>
    </div>
    <div
      style={
        Object {
          "backfaceVisibility": "hidden",
          "position": "absolute",
          "top": "0",
          "transform": "rotateY(180deg)",
          "width": "100%",
        }
      }
    >
      <div
        className="p-2"
        style={
          Object {
            "alignItems": "center",
            "border": "1px solid #f00",
            "borderRadius": "5px",
            "display": "flex",
            "height": "175px",
            "margin": "auto",
            "width": "50%",
          }
        }
      >
        test
      </div>
    </div>
  </div>
</div>
`;

exports[`FlipCard changes the class when hovered 3`] = `
<div
  className="col-xs-12 col-md-6 col-lg-3"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "height": "250px",
      "perspective": "800px",
    }
  }
>
  <div
    style={
      Object {
        "backfaceVisibility": "hidden",
        "transform": "rotateY(0deg)",
        "transformStyle": "preserve-3d",
        "transition": "transform 1s linear",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "backfaceVisibility": "hidden",
          "position": "absolute",
          "top": "0",
          "width": "100%",
        }
      }
    >
      <i
        aria-hidden="true"
        className="fa fa-5x fa-github"
        style={
          Object {
            "color": "#f00",
            "fontSize": "10em",
          }
        }
      />
      <div
        className="text-uppercase"
        style={
          Object {
            "fontSize": "1.5em",
          }
        }
      >
        Hello
      </div>
    </div>
    <div
      style={
        Object {
          "backfaceVisibility": "hidden",
          "position": "absolute",
          "top": "0",
          "transform": "rotateY(180deg)",
          "width": "100%",
        }
      }
    >
      <div
        className="p-2"
        style={
          Object {
            "alignItems": "center",
            "border": "1px solid #f00",
            "borderRadius": "5px",
            "display": "flex",
            "height": "175px",
            "margin": "auto",
            "width": "50%",
          }
        }
      >
        test
      </div>
    </div>
  </div>
</div>
`;
